cli/proc_state.rs

```diff
@@ -560,6 +595,101 @@ impl ProcState {
       None
     }
   }
+
+  /// Translates given CJS module into ESM. This function will perform static
+  /// analysis on the file to find defined exports and reexports.
+  ///
+  /// For all discovered reexports the analysis will be performed recursively.
+  ///
+  /// If successful a source code for equivalent ES module is returned.
+  async fn translate_cjs_to_esm(
+    &self,
+    specifier: &ModuleSpecifier,
+    code: String,
+    media_type: MediaType,
+  ) -> Result<String, AnyError> {
```

Comment:
Why is this a method on ProcState? I see that it uses `self.file_fetcher` but otherwise it seems independent of ProcState. Consider moving this into `cli/compat/` as a standalone function that take a file_fetcher argument.

Reply:
Done